pandas/core/indexes/datetimes.py

```diff
@@ -805,6 +647,172 @@ def intersection(self, other):
             left_chunk = left.values[lslice]
             return self._shallow_copy(left_chunk)
 
+    # --------------------------------------------------------------------
```

Comment:
can you annotate these

Reply:
These are end-of-section separators; the methods below are un-grouped.  I can collect all the groups at the top so there is only one un-annotated?